postreise/analyze/transmission/tests/test_congestion_surplus.py

```diff
@@ -1,112 +1,96 @@
-        self._check_return(expected_return, surplus)
-
-    def test_calculate_congestion_surplus_three_times(self):
-        """First: congested. Second: uncongested. Third: uncongested, fuzzy."""
-        time_indices = ["t1", "t2", "t3"]
-        demand = pd.DataFrame({"UTC": time_indices, 1: [410] * 3, 2: [0] * 3})
-        lmp = pd.DataFrame(
-            {
-                "UTC": time_indices,
-                1: [7.5, 7.5, 7.5],
-                2: [11.25, 7.5, 7.5],
-                3: [10, 7.5, 7.49],
-            }
-        )
-        pg = pd.DataFrame(
-            {
-                "UTC": time_indices,
-                "A": [50, 125, 125],
-                "B": [285] * 3,
-                "C": [0] * 3,
-                "D": [75, 0, 0],
-            }
-        )
-        for df in (demand, lmp, pg):
-            df.set_index("UTC", inplace=True)
-        mock_scenario = MockScenario(grid_attrs, demand=demand, lmp=lmp, pg=pg)
-
-        expected_return = pd.Series(
-            data=[787.5, 0, 0],
-            index=pd.date_range(start="2016-01-01", periods=3, freq="H"),
-        )
-        expected_return.rename_axis("UTC")
-
-        surplus = calculate_congestion_surplus(mock_scenario)
-        self._check_return(expected_return, surplus)
-
-
-class TestMappingHelpers(unittest.TestCase):
-    def _check_expected(self, test_return, expected_return, name):
-        self.assertIsInstance(test_return, pd.DataFrame)
-        msg = "Index mismatch for " + name
-        np.testing.assert_array_equal(test_return.index, expected_return.index, msg)
-        msg = "Column mismatch for " + name
-        np.testing.assert_array_equal(test_return.columns, expected_return.columns, msg)
-        msg = "Values mismatch for " + name
-        np.testing.assert_array_almost_equal(
-            test_return.to_numpy(), expected_return.to_numpy(), err_msg=msg
-        )
-
-    def test_map_demand_to_buses(self):
-        grid = MockGrid(grid_attrs)
-        demand = pd.DataFrame({"UTC": ["t1", "t2"], 1: [410] * 2, 2: [0] * 2})
-        demand.set_index("UTC", inplace=True)
-        expected_return = pd.DataFrame(
-            {"UTC": ["t1", "t2"], 1: [50] * 2, 2: [60] * 2, 3: [300] * 2}
-        )
-        expected_return.set_index("UTC", inplace=True)
-
-        bus_demand = map_demand_to_buses(grid, demand)
-        self._check_expected(bus_demand, expected_return, name="bus_demand")
+def _check_return(expected_return, surplus):
+    assert isinstance(surplus, pd.Series)
+    msg = "Time series indices don't match"
+    np.testing.assert_array_equal(
+        surplus.index.to_numpy(), expected_return.index.to_numpy(), msg
+    )
+    msg = "Values don't match expected"
+    np.testing.assert_array_equal(surplus.to_numpy(), expected_return.to_numpy(), msg)
+
+
+def test_calculate_congestion_surplus_single_time(monkeypatch):
+    """Congested case from Kirschen & Strbac Section 5.3.2.4"""
+
+    def mock_get_data(*args, **kwargs):
```

Comment:
We don't have keyword arguments in get_data, do we?

Reply:
I think this covers us in all cases: whether we call `get_data(scenario_info, "demand")` or `get_data(scenario_info=scenario_info, field_name="demand")`

Reply:
Agree. Although not in `get_bus_demand()`.